### ISSUE TITLE
Cannot compile on branch geometry; JSON error with env

### ISSUE THREAD

**Issue by hypotext** [penrose/penrose#217]
On branch `geometry`, commit 649e53e, when I do `stack clean` and then `stack build --profile --fast` I get the following error:

```
[ 7 of 18] Compiling Env              ( src/Env.hs, .stack-work/dist/x86_64-osx/Cabal-1.24.2.0/build/penrose/penrose-tmp/Env.p_o )
               
/Users/katherineye/Desktop/Code/penrose2/src/Env.hs:489:10: error:
    • No instance for (ToJSON T.Token)
        arising from a use of ‘aeson-1.2.3.0:Data.Aeson.Types.ToJSON.$dmtoJSON’
    • In the expression:
        aeson-1.2.3.0:Data.Aeson.Types.ToJSON.$dmtoJSON @StmtNotationRule
      In an equation for ‘toJSON’:
          toJSON
            = aeson-1.2.3.0:Data.Aeson.Types.ToJSON.$dmtoJSON @StmtNotationRule
      In the instance declaration for ‘ToJSON StmtNotationRule’
```

`git blame` says that the JSON lines came from commit 503bc30b. I doubt that any changes we made in the various branches `transforms`, `speed`, `geometry`, or `syntax` would have affected the Substance tokenizer. Seems to have something to do with changes made for `lsp`. 

@wodeni @maxkrieger Do you have an idea of why this might be happening? Perhaps you didn't check in a modified `SubstanceTokenizer` file? I am reluctant to modify `SubstanceTokenizer` to fix this change because it looks like it contains a ton of generated or pasted-in Alex code.

(Note: I'm working in a second copy of the repo so maybe there's some kind of weird package linking bug between the two repo versions)

Update: even after rebuilding from scratch in the original penrose repo this error still happens

Update: if I reset to an earlier commit, before the `syntax` merge, build there, and then checkout the newest commit from `geometry` and build, then it works. Weird!

**Comment by wodeni** [comment]
It has something to do with the renaming of the tokenizer. Make sure you are linking to the right version of the `.x` file. For `lsp`, we added Aeson support to the `Token`s, that's why `Env.hs` is looking for `ToJSON` instances.

**Comment by hypotext** [comment]
Resolved by @wodeni in 3cc2b2c.
